radiography/src/main/java/radiography/ScannableView.kt

```diff
@@ -0,0 +1,37 @@
+package radiography
+
+import android.view.View
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.IntSize
+import radiography.ScannableView.AndroidView
+import radiography.ScannableView.ComposeView
+import radiography.compose.ExperimentalRadiographyComposeApi
+
+/**
+ * Represents a logic view that can be rendered as a node in the view tree.
+ *
+ * Can either be an actual Android [View] ([AndroidView]) or a grouping of Composables that roughly
+ * represents the concept of a logical "view" ([ComposeView]).
+ */
+public sealed class ScannableView {
+
+  public class AndroidView(val view: View) : ScannableView()
+
+  /**
+   * Represents a group of Composables that make up a logical "view".
+   *
+   * @param modifiers The list of [Modifier]s that are currently applied to the Composable.
+   */
+  @ExperimentalRadiographyComposeApi
+  public class ComposeView(
+    val width: Int,
+    val height: Int,
+    val modifiers: List<Modifier>
+  ) : ScannableView() {
+
+    internal constructor(
+      size: IntSize,
+      modifiers: List<Modifier>
```

Comment:
Does this compile without a dependency on compose? ie can you load a sealed class without loading one of the subclasses?

Reply:
Yep. It's just a generic, doesn't exist at runtime. Also for the record, we have UI tests that validate that this library works when used with an older Compose version, and with no Compose dependencies at all, so if CI is green, it's usable without Compose.